zipline/testing/predicates.py

```diff
@@ -560,6 +568,18 @@ def assert_array_equal(result,
             err_msg=msg,
         )
     except AssertionError as e:
+        if is_datetime:
+            raise AssertionError(
```

Comment:
if the arrays have datetime dtype and their values do not match, we need to cast them back to datetime64 to provide a coherent assertion error.